test/suite/test_hs33.py

```diff
@@ -0,0 +1,150 @@
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def large_updates(self, session, uri, value, ds, nrows, timestamp=False):
+        cursor = session.open_cursor(uri)
+        for i in range(1, nrows):
+            if timestamp == True:
+                session.begin_transaction()
+            cursor.set_key(ds.key(i))
+            cursor.set_value(value)
+            self.assertEqual(cursor.update(), 0)
+            if timestamp == True:
+                session.commit_transaction('commit_timestamp=' + self.timestamp_str(i + 1))
+        cursor.close()
+
+    def large_modifies(self, session, uri, offset, ds, nrows, timestamp=False):
+        cursor = session.open_cursor(uri)
+        for i in range(1, nrows):
+            # Unlike inserts and updates, modify operations do not implicitly start/commit a transaction.
+            # Hence, we begin/commit transaction manually.
+            session.begin_transaction()
+            cursor.set_key(ds.key(i))
+
+            mods = []
+            mod = wiredtiger.Modify('A', offset, 1)
+            mods.append(mod)
+            self.assertEqual(cursor.modify(mods), 0)
+
+            if timestamp == True:
+                session.commit_transaction('commit_timestamp=' + self.timestamp_str(i + 1))
+            else:
+                session.commit_transaction()
+        cursor.close()
+
+    def add_insert(self, uri, ds, value, nrows):
+        session = self.session
+        cursor = session.open_cursor(uri)
+        self.pr('insert: ' + uri + ' for ' + str(nrows) + ' rows')
+        for i in range(1, nrows):
+            session.begin_transaction()
+            cursor[ds.key(i)] = value
+            session.commit_transaction()
+        cursor.close()
+
+    def test_hs_recovery(self):
+        ntables = 100
+        nrows = 5
+
+        bigvalue = b"aaaaa" * 100
+        bigvalue2 = b"ccccc" * 100
+
+        # Create a large number of tables.
+        tables = {}
+        for i in range(1, ntables):
+            uri = 'table:table' + str(i)
+            ds = SimpleDataSet(self, uri, 0, key_format='S', value_format='u',
+                               config='log=(enabled=false)')
+            ds.populate()
+            tables[uri] = ds
+
+        # Insert an initial set of data.
+        for uri, ds in tables.items():
+            self.add_insert(uri, ds, bigvalue, nrows)
+
+        # Apply some updates to move data to the history store.
+        session2 = self.conn.open_session()
+        session2.begin_transaction()
+        # Large updates with session 1.
+        for uri, ds in tables.items():
+            self.large_updates(self.session, uri, bigvalue2, ds, nrows)
+
+        self.session.checkpoint()
+        session2.rollback_transaction()
+        session2.close()
+
+        self.conn.reconfigure('timing_stress_for_test=[checkpoint_stop]')
```

Comment:
Please, explain why we are reconfiguring the connection with this timing stress.

Reply:
That should be explained here
```
            # Wait until we reach the checkpoint stop timing stress point before copying the
            # database. This point is placed before we sync the metadata file so that the snapshot
            # includes incomplete checkpoint log records.
```

Reply:
I can move it above?

Reply:
Yes, we can move it here and say that the stat that reflects that state is checked down below.

Reply:
Added https://github.com/wiredtiger/wiredtiger/pull/11857/commits/7af09d6ed18d73797653bef2c23120f8eca999f6